inst/shiny/DiagnosticsExplorer/server.R

```diff
@@ -4119,69 +2722,226 @@ shiny::shinyServer(function(input, output, session) {
       data$vocabularyVersionCdm <- "Not in data"
     }
     if (!'vocabularyVersion' %in% colnames(database)) {
       data$vocabularyVersion <- "Not in data"
     }
-    data <- data %>%
-      dplyr::select(
-        .data$databaseId,
-        .data$databaseName,
-        .data$vocabularyVersionCdm,
-        .data$vocabularyVersion,
-        .data$description
-      ) 
     
-    options = list(
-      pageLength = 100,
-      lengthMenu = list(c(10, 100, 1000, -1), c("10", "100", "1000", "All")),
-      searching = TRUE,
-      lengthChange = TRUE,
-      ordering = TRUE,
-      paging = TRUE,
-      searchHighlight = TRUE,
-      columnDefs = list(
-        list(width = "50%", targets = 4)
+    keyColumns <- intersect(
+      colnames(data),
+      c(
+        "databaseId",
+        "databaseName",
+        "vocabularyVersionCdm",
+        "vocabularyVersion",
+        "description",
+        "startTime",
+        "runTime",
+        "runTimeUnits",
+        "sourceReleaseDate",
+        "cdmVersion",
+        "cdmReleaseDate" ,
+        "observationPeriodMinDate",
+        "observationPeriodMaxDate"
       )
     )
-    sketch <- htmltools::withTags(table(class = "display",
-                                        thead(
-                                          tr(
-                                            th(rowspan = 2, "ID"),
-                                            th(rowspan = 2, "Name"),
-                                            th(
-                                              "Vocabulary version",
-                                              colspan = 2,
-                                              class = "dt-center",
-                                              style = "border-right:1px solid silver;border-bottom:1px solid silver"
-                                            ),
-                                            th(rowspan = 2, "Description")
-                                          ),
-                                          tr(lapply(
-                                            c("CDM source", "Vocabulary table"), th, style = "border-right:1px solid silver;border-bottom:1px solid silver"
-                                          ))
-                                        )))
-    table <- DT::datatable(
-      data ,
-      options = options,
-      container = sketch,
-      rownames = FALSE,
-      class = "stripe compact"
-    ) 
-    return(table)
-  }, server = TRUE)
+    
+    dataColumns <- c("personsInDatasource",
+                     "recordsInDatasource",
+                     "personDaysInDatasource")
+    
+    data <- getDisplayTableSimple(data = data,
+                                  keyColumns = keyColumns,
+                                  dataColumns = dataColumns)
+    return(data)
+  })
+  
+  #Sign in/Authentication ---------------------------------------------
+  activeLoggedInUser <- reactiveVal(NULL)
+  observeEvent(eventExpr = input$signInModelPopUp,
```

Comment:
I have a few problems with the implementation of this sign in module:

* It seems to crash the shiny app when bad details are entered

* There must always be a webapi for the shiny apps - this is problematic because that couples this entire shiny app to having some webapi instance 

* This is difficult to check and relates all cohorts to a web api instance - what if the cohorts here don't map to the web api instance anymore? 

* This security model isn't ideal in a production setting - if there is a hack there is an obvious DDOS for our production server - keep sending comments until the postgres db crashes. We have no way of preventing this on our live environments

I don't want this to sound like I don't want annotations I just want to get the security model right. I think behind an organisational firewall that the current implementation is ok (i'd actually argue for having no log in and just allowing it to be edited anonymous wiki style, perhaps the user just having to add their name) but if we allow this on data.ohdsi.org there could be significant problems that need addressing. Lee is busy enough as it is!